x/iro/types/iro.pb.go

```diff
@@ -0,0 +1,1663 @@
+	ClaimedAmt github_com_cosmos_cosmos_sdk_types.Int `protobuf:"bytes,10,opt,name=claimed_amt,json=claimedAmt,proto3,customtype=github.com/cosmos/cosmos-sdk/types.Int" json:"claimed_amt"`
+	// The incentive plan parameters for the tokens left after the plan is settled.
+	IncentivePlanParams IncentivePlanParams `protobuf:"bytes,11,opt,name=incentive_plan_params,json=incentivePlanParams,proto3" json:"incentive_plan_params"`
+}
+
+func (m *Plan) Reset()         { *m = Plan{} }
+func (m *Plan) String() string { return proto.CompactTextString(m) }
+func (*Plan) ProtoMessage()    {}
+func (*Plan) Descriptor() ([]byte, []int) {
+	return fileDescriptor_e7d27cc6b5064d3f, []int{2}
+}
+func (m *Plan) XXX_Unmarshal(b []byte) error {
+	return m.Unmarshal(b)
+}
+func (m *Plan) XXX_Marshal(b []byte, deterministic bool) ([]byte, error) {
+	if deterministic {
+		return xxx_messageInfo_Plan.Marshal(b, m, deterministic)
+	} else {
+		b = b[:cap(b)]
+		n, err := m.MarshalToSizedBuffer(b)
+		if err != nil {
+			return nil, err
+		}
+		return b[:n], nil
+	}
+}
+func (m *Plan) XXX_Merge(src proto.Message) {
+	xxx_messageInfo_Plan.Merge(m, src)
+}
+func (m *Plan) XXX_Size() int {
+	return m.Size()
+}
+func (m *Plan) XXX_DiscardUnknown() {
+	xxx_messageInfo_Plan.DiscardUnknown(m)
+}
+
+var xxx_messageInfo_Plan proto.InternalMessageInfo
+
+func (m *Plan) GetId() uint64 {
+	if m != nil {
+		return m.Id
+	}
+	return 0
+}
+
+func (m *Plan) GetRollappId() string {
+	if m != nil {
+		return m.RollappId
+	}
+	return ""
+}
+
+func (m *Plan) GetModuleAccAddress() string {
+	if m != nil {
+		return m.ModuleAccAddress
+	}
+	return ""
+}
+
+func (m *Plan) GetTotalAllocation() types.Coin {
+	if m != nil {
+		return m.TotalAllocation
+	}
+	return types.Coin{}
+}
+
+func (m *Plan) GetBondingCurve() BondingCurve {
+	if m != nil {
+		return m.BondingCurve
+	}
+	return BondingCurve{}
+}
+
+func (m *Plan) GetSettledDenom() string {
+	if m != nil {
+		return m.SettledDenom
+	}
+	return ""
+}
+
+func (m *Plan) GetStartTime() time.Time {
+	if m != nil {
+		return m.StartTime
+	}
+	return time.Time{}
+}
+
+func (m *Plan) GetPreLaunchTime() time.Time {
+	if m != nil {
+		return m.PreLaunchTime
+	}
+	return time.Time{}
+}
+
+func (m *Plan) GetIncentivePlanParams() IncentivePlanParams {
+	if m != nil {
+		return m.IncentivePlanParams
+	}
+	return IncentivePlanParams{}
+}
+
+type IncentivePlanParams struct {
+	// start_time_after_settlement is the time after IRO settlement when the
+	// distribution of the remaining tokens as incentives will start
+	StartTimeAfterSettlement time.Duration `protobuf:"bytes,1,opt,name=start_time_after_settlement,json=startTimeAfterSettlement,proto3,stdduration" json:"start_time_after_settlement"`
+	// num_epochs_paid_over is the number of total epochs distribution will be
+	// completed over
+	NumEpochsPaidOver uint64 `protobuf:"varint,2,opt,name=num_epochs_paid_over,json=numEpochsPaidOver,proto3" json:"num_epochs_paid_over,omitempty"`
```

Comment:
those params descriptions are misleading as they are not de-facto what the plan will enforce, but are boundries (i.e min)